src/libzfs/py_zfs.c

```diff
@@ -510,6 +510,221 @@ PyObject *py_zfs_iter_root_datasets(PyObject *self,
 	Py_RETURN_FALSE;
 }
 
+PyDoc_STRVAR(py_zfs_test_topology__doc__,
+"test_topology(*, topology) -> None\n\n"
+"--------------------------------------\n\n"
+"Creates a VDEV Tree from given topology and prints the nvlist to stdout.\n"
+"This can be helpful while debugging issues in VDEV tree creation."
+"Parameters\n"
+"----------\n"
+"topology: List of Dictionaries\n"
+"    Topology is a list of Dictionaries where each item of the list would\n"
+"    specify the formation of each VDEV.\n"
+"        \'root\': Str\n"
+"            this key should specify the class of VDEV. This can be any of\n"
+"            \'DATA\', \'LOG\', \'DEDUP\', \'SPECIAL\', \'SPARE\',or \'CACHE\'.\n"
+"        \'type\': Str\n"
+"            specify the type of VDEV, which can be any one of \'MIRROR\',\n"
+"            \'STRIPE\', \'RAIDZ1\', \'RAIDZ2\', \'RAIDZ3\', \'DRAID1\'\n"
+"            \'DRAID2\' or \'DRIAD3\'.\n"
+"        \'devices\': List of Str\n"
+"            List of valid paths in string format for disks.\n\n"
+"Returns\n"
+"-------\n"
+"None\n\n"
+"Raises:\n"
+"-------\n"
+"TypeError:\n"
+"    If there is an issue while parsing the topology, TypeError will be raised.\n"
+);
+static
+PyObject *py_zfs_test_topology(PyObject *self, PyObject *args,
```

Comment:
I think it would be use the `py_dump_nvlist` function I wrote (see `common/nvlist_utils.c`) to convert the nvlist to a string and return it. This is more useful for python devs who may not expect stuff to be written to stdout.

Reply:
If you want raw nvlist_dump you can:
```
out = py_dump_nvlist(tree, B_FALSE);
```

JSON
```
out = py_dump_nvlist(tree, B_TRUE);
```

Reply:
Thanks, updated to use `py_dump_nvlist`.